features/integration/c2c.feature

```diff
@@ -0,0 +1,115 @@
+
+    Then I build the transaction group with the composer. If there is an error it is "".
+    Then I gather signatures with the composer.
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    And The app should have returned ABI types "(uint64,byte[17]):(uint64,byte[17])".
+    And The 0th atomic result for randomInt(42) proves correct
+    And The 1th atomic result for randomInt(1337) proves correct
+    And I can retrieve all inner transactions that were called from the atomic transaction with call graph "['randInt(uint64)(uint64,byte[17])','randInt(uint64)(uint64,byte[17])']".
+
+    ###### ------ app at context index 1: RandomByte ------ ######
+    When I build an application transaction with the transient account, the current application, suggested params, operation "create", approval-program "programs/random_byte.teal", clear-program "programs/one.teal.tok", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    Given I remember the new application ID.
+    # Need to fund RandomByte because it pays for calling RandomInteger:
+    And I fund the current application's address with 100000000 microalgos.
+
+    Given a new AtomicTransactionComposer
+
+    # randElement("hello",RandomByte) -> (c, witnessString)
+    Given I add the nonce "Thing One"
+    When I create the Method object from method signature "randElement(string,application)(byte,byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "AAVoZWxsbw==,ctxAppIdx:0" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # randElement("goodbye",RandomByte) -> (c, witnessString)
+    Given I add the nonce "Thing Two"
+    When I create the Method object from method signature "randElement(string,application)(byte,byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "AAdnb29kYnll,ctxAppIdx:0" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    Then I build the transaction group with the composer. If there is an error it is "".
+
+    Then I gather signatures with the composer.
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    And The app should have returned ABI types "(byte,byte[17]):(byte,byte[17])".
```

Comment:
Does the "ABI types" string need to be repeated?
```suggestion
    And The app should have returned ABI types "(byte,byte[17])".
```

Reply:
Yes, because there are 2 separate calls to `randElement()` inside of the atomic transactions, so their responses should be at corresponding indexes. I guess it's possible to simplify the test to only handle a call of the same method again and again, and assume that the responses all have the same type, but that would break the python test as currently written.

Reply:
Got it, makes sense